bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSBridgeHandler.java

```diff
@@ -0,0 +1,362 @@
+            initialConnection.cancel(false);
+            this.initialConnection = null;
+        }
+    }
+
+    @Override
+    public void onEvent(BaseMessage message) {
+        if (message instanceof VersionMessage event) {
+            Map<String, String> properties = new HashMap<>();
+            properties.put(BindingConstants.PROPERTY_DRIVER_VERSION, event.driverVersion);
+            properties.put(BindingConstants.PROPERTY_SERVER_VERSION, event.serverVersion);
+            properties.put(BindingConstants.PROPERTY_SCHEMA_MIN, String.valueOf(event.minSchemaVersion));
+            properties.put(BindingConstants.PROPERTY_SCHEMA_MAX, String.valueOf(event.maxSchemaVersion));
+            properties.put(BindingConstants.PROPERTY_HOME_ID, String.valueOf(event.homeId));
+            this.getThing().setProperties(properties);
+            return;
+        }
+
+        if (message instanceof ResultMessage result) {
+            if (result.messageId != null && result.messageId.startsWith("getvalue|")) {
+                Event event = createEventFromMessageId(result.messageId,
+                        result.result != null ? result.result.value : null);
+                if (event == null) {
+                    return;
+                }
+
+                ZwaveNodeListener nodeListener = nodeListeners.get(event.nodeId);
+                if (nodeListener != null) {
+                    nodeListener.onNodeStateChanged(event);
+                }
+                return;
+            }
+            if (result.result == null || result.result.state == null) {
+                logger.debug("ResultMessage missing result or state, ignoring.");
+                return;
+            }
+            procesStateUpdate(result.result.state);
+            updateStatus(ThingStatus.ONLINE);
+            return;
+        }
+
+        if (message instanceof EventMessage eventMsg && eventMsg.event != null) {
+            String eventType = eventMsg.event.event;
+            ZwaveNodeListener nodeListener = nodeListeners.get(eventMsg.event.nodeId);
+            switch (eventType) {
+                case "value updated":
+                case "value notification":
+                    if (nodeListener != null) {
+                        nodeListener.onNodeStateChanged(eventMsg.event);
+                    }
+                    break;
+                case "alive":
+                    if (nodeListener != null) {
+                        nodeListener.onNodeAlive(eventMsg.event);
+                    }
+                    break;
+                case "dead":
+                    if (nodeListener != null) {
+                        nodeListener.onNodeDead(eventMsg.event);
+                    }
+                    break;
+                default:
+                    logger.trace("Unhandled event type: {}", eventType);
+            }
+            return;
+        }
+    }
+
+    public @Nullable Event createEventFromMessageId(String messageId, @Nullable Object value) {
```

Comment:
Do you need a public method ?

Reply:
Looks like you did not answer to me comment.

Reply:
Yeah, sorry, this answer ended up in the wrong comment.. I have to look into this modifier later today as i don;t have access to my pc now.

Reply:
marked private